llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -1398,6 +1392,54 @@ DebugAbbrevOffsetsRefAdaptor::encodeOffsets(ArrayRef<size_t> Offsets) {
   return EncodedOffsets;
 }
 
+MachObjectWriter::AddendsSizeAndOffset
+InMemoryCASDWARFObject::zeroStmtListInCU(MutableArrayRef<char> DebugInfoData,
+                                         uint64_t AbbrevOffset,
+                                         uint64_t CUOffset) {
+  uint32_t StmtListOffset;
+  uint8_t StmtListNumBytes;
+  uint64_t Fixup;
+  DWARFUnitVector UV;
+  uint64_t Address = 0;
+  DWARFSection Section = {toStringRef(DebugInfoData), Address};
+  DWARFUnitHeader Header;
+  DWARFDebugAbbrev Abbrev;
+  StringRef AbbrevSectionContribution =
+      getAbbrevSection().drop_front(AbbrevOffset);
+  Abbrev.extract(DataExtractor(AbbrevSectionContribution, isLittleEndian(), 8));
+  auto DWARFObj = std::make_unique<InMemoryCASDWARFObject>(*this);
+  auto DWARFContextHolder = std::make_unique<DWARFContext>(std::move(DWARFObj));
+  auto *DWARFCtx = DWARFContextHolder.get();
+  uint64_t offset_ptr = 0;
+  Header.extract(*DWARFCtx,
+                 DWARFDataExtractor(*this, Section, isLittleEndian(), 8),
+                 &offset_ptr, DWARFSectionKind::DW_SECT_INFO);
+
+  DWARFCompileUnit U(*DWARFCtx, Section, Header, &Abbrev, &getRangesSection(),
+                     &getLocSection(), getStrSection(), getStrOffsetsSection(),
+                     &getAddrSection(), getLocSection(), isLittleEndian(),
+                     false, UV);
+
+  bool IsStmtListOffsetFound = false;
+  if (DWARFDie CUDie = U.getUnitDIE(false)) {
+    for (const DWARFAttribute &AttrValue : CUDie.attributes()) {
+      if (AttrValue.Attr == llvm::dwarf::DW_AT_stmt_list) {
```

Comment:
To summarize an offline conversation: If we look at `AttrValue.Value.getForm()` instead we can generalize this and just take a DenseSet if interesting FORMs we want to stash in the CASFixup block.

https://llvm.org/doxygen/classllvm_1_1DWARFFormValue.html